src/shared/lib/index.ts

```diff
@@ -336,3 +336,22 @@ export function arrayFromRange<Element = number>(
     return cb(number);
   });
 }
+
+/**
+ * Intersperse array elements with a separator.
+ *
+ * @param collection
+ * @param separator - element placed between collection members
+ * @returns list of interspersed elements
+ */
+export function intersperse<A, B>(collection: A[], separator: B): Array<A | B> {
+  return reduce(
+    collection,
+    (acc: Array<A | B>, val: A, index: number) => [
+      ...acc,
+      ...(index > 0 ? [separator] : []),
+      val
+    ],
+    []
+  );
+}
```

Comment:
I recommend using only one type parameter here: `intersperse<A>(collection: A[], separator: A): A[]`. This keeps the function simple and is more type-safe.

If you need the `A | B` functionality, you can still set the type parameter at the call site:

```ts
const foo = intersperse<ReactElement | string>([<div></div>], "sep");
```

Reply:
K I think I had a misunderstanding about the `Intersperse` API we decided on. Looking at the usage again, what you're suggesting makes sense so I'll change it.